### ISSUE TITLE
core: all purpose modal view

### ISSUE THREAD

**Issue by Augustyniak** [lyft/clutch#415]
A control that allows to define a view that's displayed modally - on top of the existing screen/page. 

<img width="986" alt="Screen Shot 2020-09-15 at 7 19 10 PM" src="https://user-images.githubusercontent.com/4410346/93284880-7cd26e80-f788-11ea-954f-a02ce15c8b83.png">

This issue is related to https://github.com/lyft/clutch/issues/5

Complexity [S/M/L]:
S

**Comment by danielhochman** [comment]
I put a request for the designer to mock one up.

In the meantime have you played at all with https://material-ui.com/components/dialogs/?

**Comment by danielhochman** [comment]
Here are the new designs:

![Screenshot from 2020-09-22 12-15-28](https://user-images.githubusercontent.com/4712430/93915257-5dc66600-fccd-11ea-8ec9-8259b305c746.png)

Once we get Storybook up and running (hopefully this week) we will be able to introduce this component as it appears above.

**Comment by Augustyniak** [comment]
Thanks for sharing these - they look nice. 

Frankly speaking, I'm a little bit hesitant to adding new custom components to Clutch since I ran into multiple styling related issue when I tried to add my own datetime picker (somehow related to https://github.com/lyft/clutch/issues/414) or when I tried to add a custom implementation of `Select` (related to https://github.com/lyft/clutch/issues/433). 

@dschaller did make a few styling related fixes in Clutch so maybe my worries are not valid anymore but ideally I would prefer to use a Clutch provided control here instead of introducing a custom one.
